source/iterator/cdc.go

```diff
@@ -0,0 +1,236 @@
+		return fmt.Errorf("process updated items: %w", err)
+	}
+
+	return nil
+}
+
+// processUpdatedItems retrieves items that were updated after the provided timestamp.
+func (c *CDC) processUpdatedItems(ctx context.Context, updatedAfter time.Time) error {
+	if timestampResource, ok := hubspot.TimestampResources[c.resource]; ok {
+		return c.fetchTimestampBasedItems(ctx, timestampResource, updatedAfter)
+	}
+
+	if searchResource, ok := hubspot.SearchResources[c.resource]; ok {
+		return c.fetchSearchBasedItems(ctx, searchResource, updatedAfter)
+	}
+
+	return nil
+}
+
+// fetchTimestampBasedItems fetches items by filtering them by updatedAt field.
+func (c *CDC) fetchTimestampBasedItems(
+	ctx context.Context, resource hubspot.TimestampResource, updatedAfter time.Time,
+) error {
+	listOpts := &hubspot.ListOptions{
+		Limit:        c.bufferSize,
+		UpdatedAfter: &updatedAfter,
+		Sort:         hubspot.UpdatedAtListSortKey,
+	}
+
+	listResp, err := c.hubspotClient.List(ctx, c.resource, listOpts)
+	if err != nil {
+		return fmt.Errorf("list items: %w", err)
+	}
+
+	for _, item := range listResp.Results {
+		err = c.routeItem(item, resource.CreatedAtFieldName, resource.UpdatedAtFieldName, updatedAfter)
+		if err != nil {
+			return fmt.Errorf("route timestamp based item: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// fetchSearchBasedItems fetches items using search endpoint.
+func (c *CDC) fetchSearchBasedItems(
+	ctx context.Context, resource hubspot.SearchResource, updatedAfter time.Time,
+) error {
+	listResp, err := c.hubspotClient.SearchByUpdatedAfter(ctx, c.resource, updatedAfter, c.bufferSize)
+	if err != nil {
+		return fmt.Errorf("list items: %w", err)
+	}
+
+	for _, item := range listResp.Results {
+		err = c.routeItem(item, resource.CreatedAtFieldName, resource.UpdatedAtFieldName, updatedAfter)
+		if err != nil {
+			return fmt.Errorf("route search based item: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// routeItem retrives createdAt and updatedAt fields from the item, compares them
+// and based on the result of the comparison decides to send a Create or Update sdk.Record.
+func (c *CDC) routeItem(
+	item hubspot.ListResponseResult, createdAtFieldName, updatedAtFieldName string, updatedAfter time.Time,
+) error {
+	itemCreatedAt, err := item.GetTimeField(createdAtFieldName)
+	if err != nil {
+		return fmt.Errorf("get item's creation date: %w", err)
+	}
+
+	itemUpdatedAt, err := item.GetTimeField(updatedAtFieldName)
+	if err != nil {
+		return fmt.Errorf("get item's update date: %w", err)
+	}
+
+	metadata := make(sdk.Metadata)
+	metadata.SetCreatedAt(itemCreatedAt)
+
+	c.position = &Position{
+		Mode: CDCPositionMode,
+		// set the timestamp to the item's updatedAt
+		// as we sort items by their updatedAt values.
+		Timestamp: &itemUpdatedAt,
+	}
+
+	sdkPosition, err := c.position.MarshalSDKPosition()
+	if err != nil {
+		return fmt.Errorf("marshal sdk position: %w", err)
+	}
+
+	if itemCreatedAt.After(updatedAfter) {
```

Comment:
maybe we can add comment to explain why we do it

Reply:
I've added the comment explaining this behavior.